### PR TITLE
system-variables: add description for `tidb_default_string_match_selectivity`

### PR DESCRIPTION
**Pull request by time-and-fate**
### What is changed, added or deleted? (Required)

As the title says.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v6.2 (TiDB 6.2 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tidb/pull/36210

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by Oreoxmt** [comment]
@time-and-fate Could you please invite a tech review? Thanks~

File: system-variables.md [hunk 1]
```diff
@@ -587,6 +587,18 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 - 范围：`[1, 256]`
 - 这个变量用来设置 DDL 操作 `re-organize` 阶段的并发度。
 
+### `tidb_default_string_match_selectivity` <span class="version-mark">从 v6.2 版本开始引入</span>
```
Comment (Oreoxmt) [inline system-variables.md:590]:
```suggestion
### `tidb_default_string_match_selectivity` <span class="version-mark">从 v6.2.0 版本开始引入</span>
```

File: system-variables.md [hunk 2]
```diff
@@ -587,6 +587,18 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 默认值：`0.8`
+- 范围：`[0, 1]`
+- 这个变量用来设置过滤条件中的 `like`, `rlike`, `regexp` 函数在行数估算时的默认选择率，以及是否对这些函数启用 TopN 辅助估算。
```
Comment (Oreoxmt) [inline system-variables.md:596]:
```suggestion
- 这个变量用来设置过滤条件中的 `like`、`rlike`、`regexp` 函数在行数估算时的默认选择率，以及是否对这些函数启用 TopN 辅助估算。
```

File: system-variables.md [hunk 3]
```diff
@@ -587,6 +587,18 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
+- TiDB 总是会尝试利用统计信息对过滤条件中的 `like` 进行估算，但是当 `like` 匹配的字符串太复杂时，或者面对 `rlike` 或 `regexp` 时，往往无法充分利用统计信息，转而使用 `0.8` 作为选择率，造成行数估算的误差较大。
+- 该变量可以用于修改这个行为，当变量被设为 `0` 以外的值时，会使用变量的值而不是默认的 `0.8` 作为选择率。
+- 如果将该变量的值设为 `0`，TiDB 在对上述三个函数进行行数估算时，会尝试利用统计信息中的 TopN 进行求值来提高估算精度，同时也会考虑统计信息中的 NULL 数。求值操作预计会造成少量性能损耗。
+- 当该变量的值被设为默认值以外的值的时候，会对 `not like`, `not rlike`, `not regexp` 的行数估算也进行相应的调整。
```
Comment (Oreoxmt) [inline system-variables.md:600]:
```suggestion
- 当该变量的值被设为默认值以外的值的时候，会对 `not like`、`not rlike`、`not regexp` 的行数估算也进行相应的调整。
```

**Comment by time-and-fate** [comment]
> @time-and-fate Could you please invite a tech review? Thanks~

Sure! @Reminiscent will review this PR.

File: system-variables.md [hunk 4]
```diff
@@ -587,6 +587,18 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
 - 范围：`[1, 256]`
 - 这个变量用来设置 DDL 操作 `re-organize` 阶段的并发度。
 
+### `tidb_default_string_match_selectivity` <span class="version-mark">从 v6.2.0 版本开始引入</span>
```
Comment (Reminiscent) [inline system-variables.md:590]: 是否需要提一下前缀条件，比如统计信息得是 version2 这样的情况

**Review by Reminiscent — APPROVED** [review]

File: system-variables.md (hunk 4 above)
Comment (time-and-fate) [reply system-variables.md:590]: 已更新。

File: system-variables.md [hunk 5]
```diff
@@ -587,6 +587,18 @@ MPP 是 TiFlash 引擎提供的分布式计算框架，允许节点之间的数
...
+
+- 作用域：SESSION | GLOBAL
+- 是否持久化到集群：是
+- 默认值：`0.8`
+- 范围：`[0, 1]`
+- 这个变量用来设置过滤条件中的 `like`、`rlike`、`regexp` 函数在行数估算时的默认选择率，以及是否对这些函数启用 TopN 辅助估算。
+- TiDB 总是会尝试利用统计信息对过滤条件中的 `like` 进行估算，但是当 `like` 匹配的字符串太复杂时，或者面对 `rlike` 或 `regexp` 时，往往无法充分利用统计信息，转而使用 `0.8` 作为选择率，造成行数估算的误差较大。
+- 该变量可以用于修改这个行为，当变量被设为 `0` 以外的值时，会使用变量的值而不是默认的 `0.8` 作为选择率。
+- 如果将该变量的值设为 `0`，TiDB 在对上述三个函数进行行数估算时，会尝试利用统计信息中的 TopN 进行求值来提高估算精度，同时也会考虑统计信息中的 NULL 数。求值操作预计会造成少量性能损耗。这个功能生效的前提是统计信息是在 [tidb_analyze_version](#tidb_analyze_version-从-v510-版本开始引入) 设为 `2` 时收集的。
```
Comment (TomShawn) [inline system-variables.md:599]:
```suggestion
- 如果将该变量的值设为 `0`，TiDB 在对上述三个函数进行行数估算时，会尝试利用统计信息中的 TopN 进行求值来提高估算精度，同时也会考虑统计信息中的 NULL 数。求值操作预计会造成少量性能损耗。这个功能生效的前提是统计信息是在 [`tidb_analyze_version`](#tidb_analyze_version-从-v510-版本开始引入) 设为 `2` 时收集的。
```

**Review by TomShawn — APPROVED** [review]
rest LGTM

**Review by Oreoxmt — APPROVED** [review]
LGTM

**Pull request merged**